test/Microsoft.Framework.DependencyInjection.Tests/ServiceProviderExtensionsTest.cs

```diff
@@ -65,6 +79,16 @@ public void ThrowsWhenNoServicesAreRegistered()
         }
 
         [Fact]
+        public void ThrowsWhenNoServicesAreRegistered_NonGeneric()
+        {
+            // Arrange
+            var serviceProvider = CreateFooServiceProvider(0);
+
+            // Act + Assert
+            Assert.Throws<InvalidOperationException>(() => serviceProvider.GetRequiredServices(typeof(IFoo)));
```

Comment:
Could you check the exception message here (so we know it's the `InvalidOperationException` you're throwing and not something else)?